pkg/app/secrets/local.go

```diff
@@ -52,3 +52,26 @@ func (sm *FromFileSecretManager) OAuth2ClientID() string {
 func (sm *FromFileSecretManager) OAuth2ClientSecret() string {
 	return sm.ClientSecret
 }
+
+type CredentialType struct {
+	Type string `json:"type"`
+}
+
+// Check if the credential type is service accout,
+// if so: return credential content
+// else: return nil
+func GetServiceAccountCredential(path string) ([]byte, error) {
```

Comment:
I think the feature getting service account credential is bound into `UnixSMType`. What about doing this logic in `NewFromFileSecretManager`, storing credential in the struct of `FromFileSecretManager`, and retrieving the credential by calling a new method of `FromFileSecretManager` like `OAuth2ClientID()` or `OAuth2ClientSecret()` does?

Reply:
Sounds good, but I think it would be better to define a standalone function at `FromFileSecretManager` rather than extending the `SecretManager` interfaces. The reason is that this service account credentials should not be used for other secret manger types.